### PR TITLE
fix deletion of effect chain presets created in current session

### LINKED ISSUE #11864: can not delete custom effect chain
1. add "Rename preset" action
* Just created a new chain and saved it via "Save as new preset"
* want to remove and effect and overwrite the preset
* = not possible via the effect chain preset menu

not possible via menu, so I go to
* Preferences -> Effects
* delete the chain from Quick Effect preset list
* try to delete chain from Effect Chain Preset list
> Preset XY is read-only and cannot be deleted.

2. fix Effects preferences

IIRC I touched the code in #10859, will take a look.

### PR DESCRIPTION
**Pull request by ronso0**
* initialize `readOnly` correctly (false) for chain presets just created via the preset menu, regression from https://github.com/mixxxdj/mixxx/pull/10859
* add "Rename preset" action to preset menu

- [x] fix occasional issue in Preferences -> Effects:
clicking any of the preset action buttons (Rename, Delete, ..) does sometimes clear the selection in the list views above, i.e. the buttons won't do anything.
Looks like the events are processed in unpredictable order: sometimes the Delete button gets focus and the list focus as well as the selection is cleared, sometimes the selection is cleared after the Delete dialog is closed (aborted, so the selected item is still in the list and selection should not change

### CODE REVIEW

**Comment by ronso0** [comment]
This can be merged as is, though I need to figure what's wrong with the focus behaviour / event processing in that dialog.

Can anyone else reproduce this (I'm using Qt 5.12.8 on Ubuntu 20.04):
* open Pref -> Effects
* select one chain preset in the Quick Effect Chain Presets list
* click Delete
* list selection is cleared, Delete button does nothing

**Comment by ronso0** [comment]
Done, ready for review.

File: src/preferences/dialog/dlgprefeffects.cpp [hunk 1]
```diff
@@ -205,19 +205,20 @@ void DlgPrefEffects::effectsTableItemSelected(const QModelIndex& selected) {
...
     VERIFY_OR_DEBUG_ASSERT(m_pFocusedChainList) {
         return;
     }
     // Clear the info box and return if the index is invalid, e.g. after clearCurrentIndex()
     // in eventFilter()
-    if (!selected.isValid() ||
-            m_pFocusedChainList->selectionModel()->selectedRows(0).count() > 1) {
+    auto* pSelModel = m_pFocusedChainList->selectionModel();
+    auto selIndices = pSelModel->selectedIndexes();
+    if (!pSelModel->hasSelection() || selIndices.count() > 1) {
```
Comment (Swiftb0y) [inline src/preferences/dialog/dlgprefeffects.cpp:216]: why `||`?

Comment (ronso0) [reply src/preferences/dialog/dlgprefeffects.cpp:216]: This should clear the info labels `if (selIndices.count() != 1)`.

I pushed the simplified version. Thanks!

**Comment by Holzhaus** [comment]
There are merge conflicts now.

**Comment by ronso0** [comment]
Conflicts resolved.

In case the new tr string is an issue I can remove 46f2f91b033ee97d82872cb1f3cff30505c10f3e and open a separate PR against `main`.

**Comment by ronso0** [comment]
_pa-ping.._

**Review by daschuer — APPROVED** [review]
LGTM and still works! Thank you.

**Pull request merged**